containers/ecr-viewer/src/app/view-data/utils/fhir-data-utils.ts

```diff
@@ -0,0 +1,72 @@
+import { FhirResource, Period } from "fhir/r4";
+
+import { evaluateOne } from "@/app/utils/evaluate";
+import { FhirPath, PathTypes } from "@/app/utils/evaluate/fhir-paths";
+
+const ONGOING_DATE = new Date("9999-01-01");
+
+const parseDate = (dateString: string | undefined): Date | undefined =>
+  dateString ? new Date(dateString) : undefined;
+
+const getDates = <K extends keyof PathTypes>(
+  obs: FhirResource,
+  datePath: FhirPath<K>,
+) => {
+  const dateElement = evaluateOne(obs, datePath) as Period | string;
```

Comment:
I think you could instead limit `FhirPathKeys` to those keys that result in a `Period | string` so that this is inherent to the function - I just did something like that with the `evaluateOneReference`